File: Backend/SOS-Clean-House-back/src/main/java/com/SOS_Clean_House_back/model/Prestador.java
package com.SOS_Clean_House_back.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@AllArgsConstructor @NoArgsConstructor @Data
public class Prestador extends Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private Integer tempoExperiencia;
    private byte[] antecedentesCriminais;
    private byte[] foto;
    private byte[] documentos;
    private String sobreMim;

    @JsonIgnore
    @OneToMany(mappedBy = "prestador")
    private List<AtividadesDesenvolvidas> atividadesDesenvolvidas;

    @JsonIgnore
    @OneToMany(mappedBy = "prestador")
    private List<DiaCalendario> diaCalendarios;

    @OneToOne(mappedBy = "prestador", cascade = CascadeType.ALL)
    private ContaPrestador contaPrestador;

    @JsonIgnore
    @OneToMany(mappedBy = "prestador")
    private List<Horario> horarios;


}

File: Frontend/SOS-Clean-House-front/src/app/componentes/interface/Prestador.ts
import { Agendamento } from "./Agendamento";
import { AtividadesDesenvolvidas } from "./AtividadesDesenvolvidas";
import { ContaPrestador } from "./ContaPrestador";
import { DiaCalendario } from "./DiaCalendario";
import { Horario } from "./Horario";
import { Usuario } from "./Usuario";

export interface Prestador extends Usuario {
    id: number;
    tempoExperiencia: number;
    antecedentesCriminais: Uint8Array | null; 
    foto: Uint8Array| null; 
    documentos: Uint8Array | null;
    sobreMim: string;
  
    agendamentos: Agendamento[];
    atividadesDesenvolvidas: AtividadesDesenvolvidas[];
    diaCalendarios: DiaCalendario[];
    contaPrestador: ContaPrestador;
    horarios: Horario[];
  }